### PR TITLE
Feature #10 -  Social buttons and read time

### LINKED ISSUE #10: Social Sharing Buttons and Reading time of Article
- [x] Social Sharing Buttons (Facebook, Twitter, Linkedin, etc)
 
- [x] Reading time of an Article

### PR DESCRIPTION
**Pull request by kennyhei**
Added social sharing buttons using Fontawesome icons and Sass.

Fixes #10

### CODE REVIEW

**Comment by roughike** [comment]
@kennyhei I edited your description a little.

If you say `Fixes #123`, the corresponding issue will automatically get closed when the pull request gets merged. :)

**Review by ragmha — CHANGES_REQUESTED** [review]
Good work! 🎉 

Few changes required:

1)  Move FontAwesome inline-style to SocialLinks.module.scss
https://github.com/CodemateLtd/codemate-blog/pull/13/files#diff-f7e02c369b70c795cab9fdae3d761de4

2) Move the timeToRead outside render method, so its a method of the BlogPostTemplate https://github.com/CodemateLtd/codemate-blog/pull/13/files#diff-b334c7299860c69226f3d8a8f4a36c1f

3) Is it possible to move all the blog-post inline-styling to an index.scss or should I work on it? https://github.com/CodemateLtd/codemate-blog/pull/13/files#diff-b334c7299860c69226f3d8a8f4a36c1f

4) Could you make the Sharing button be something like this using flex-box instead of table, https://www.behance.net/gallery/9665081/Share-Button

File: src/components/share/links/ShareLink.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  const shareMedia = `share${media[0].toUpperCase() + media.slice(1)}`
+
+  return (
+    <li className={`${styles.shareLink} ${styles[shareMedia]}`}>
+      <a target={newWindow ? '_blank' : '_self'} href={url}>
+        <FontAwesome
+          className={styles.shareIcon}
+          name={media}
+          size="2x"
+          style={{ textShadow: '0 1px 0 rgba(0, 0, 0, 0.1)' }}
```
Comment (ragmha) [inline src/components/share/links/ShareLink.js]: Move inline styles to ../SocialLinks.module.scss

File: src/templates/blog-post.js [hunk 2]
```diff
@@ -4,17 +4,36 @@ import Link from 'gatsby-link'
...
 
 import Bio from '../components/Bio'
+import SocialLinks from '../components/share/SocialLinks'
 import { rhythm, scale } from '../utils/typography'
 
 class BlogPostTemplate extends React.Component {
   render() {
     const post = this.props.data.markdownRemark
+
+    const timeToRead = () => {
```
Comment (ragmha) [inline src/templates/blog-post.js]: functions shouldn't be a part of render method, you can it a method of BlogPlostTemplate class. 

More info: https://stackoverflow.com/questions/24718709/reactjs-does-render-get-called-any-time-setstate-is-called?answertab=votes#tab-top

File: src/components/share/SocialLinks.module.scss [hunk 3]
```diff
@@ -0,0 +1,49 @@
+.shareList {
```
Comment (ragmha) [inline src/components/share/SocialLinks.module.scss]: Would be a good idea to use Flexbox layout.

A clone of this: https://www.behance.net/gallery/9665081/Share-Button would be nice, but the design will be changed in future 

More info: https://css-tricks.com/snippets/css/a-guide-to-flexbox/

**Comment by kennyhei** [comment]
Sure, I'll work on it. You can handle moving those blog-post inline-styles to index.scss if you want. I'll focus on flex-box

File: src/components/share/links/FacebookLink.js [hunk 4]
```diff
@@ -1,9 +1,11 @@
 import React from 'react'
 import ShareLink from './ShareLink'
 
-const FacebookLink = ({ slug, title, styles }) => {
+const FacebookLink = ({ pageConfig }) => {
+  const { host, port, slug, title } = pageConfig
+
   const fbLink = 'http://www.facebook.com/sharer.php',
-    shareSite = encodeURIComponent(`https://codemate.com${slug}`),
+    shareSite = encodeURIComponent(`https://${host}:${port}${slug}`),
```
Comment (roughike) [inline src/components/share/links/FacebookLink.js]: The Netlify builds broke due to the following reason: 

```
GraphQL Error: There was an error while compiling your site's GraphQL queries.
Invariant Violation: GraphQLParser: Unknown field `host` on type `Site`. Source: document `BlogPostBySlug` file: `GraphQL request`.
```

I would bet that these will also break "real" sites (think when this blog gets published) as well, although locally on `localhost:4000` they will work fine.

File: src/components/share/links/FacebookLink.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import ShareLink from './ShareLink'
+
+const FacebookLink = ({ pageConfig }) => {
+  const { host, port, slug, title } = pageConfig
+
+  const fbLink = 'http://www.facebook.com/sharer.php',
+    shareSite = encodeURIComponent(`https://${host}:${port}${slug}`),
```
Comment (roughike) [inline src/components/share/links/FacebookLink.js]: The Netlify builds broke due to the following reason: 

```
GraphQL Error: There was an error while compiling your site's GraphQL queries.
Invariant Violation: GraphQLParser: Unknown field `host` on type `Site`. Source: document `BlogPostBySlug` file: `GraphQL request`.
```

I would bet that these will also break "real" sites (think when this blog gets published) as well, although locally on `localhost:4000` they will work fine.

File: src/components/share/links/LinkedinLink.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import ShareLink from './ShareLink'
+
+const LinkedinLink = ({ pageConfig }) => {
+  const { host, port, slug, title } = pageConfig
+
+  const linkedinLink = 'http://linkedin.com/shareArticle',
+    shareSite = encodeURIComponent(`https://${host}:${port}${slug}`),
```
Comment (roughike) [inline src/components/share/links/LinkedinLink.js]: See the very first comment above.

File: src/components/share/links/MailLink.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import ShareLink from './ShareLink'
+
+const MailLink = ({ pageConfig }) => {
+  const { host, port, slug, title } = pageConfig
+
+  const mailLink = 'mailto:',
+    shareSite = encodeURIComponent(`https://${host}:${port}${slug}`),
```
Comment (roughike) [inline src/components/share/links/MailLink.js]: See the very first comment above.

File: src/components/share/links/TwitterLink.js [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import ShareLink from './ShareLink'
+
+const TwitterLink = ({ pageConfig }) => {
+  const { host, port, slug, title } = pageConfig
+
+  const tweetLink = 'https://twitter.com/share',
+    shareSite = encodeURIComponent(`https://${host}:${port}${slug}`),
```
Comment (roughike) [inline src/components/share/links/TwitterLink.js]: See the very first comment above.

**Review by roughike — CHANGES_REQUESTED** [review]
I think the latest changes broke the build for deployed sites. In other words, it works locally, but will not work when trying to actually publish the blog.

**Review by ragmha — COMMENTED** [review]
Regarding the Social Links there seems to be a lot of repeated code.A good reference should be this https://github.com/tfiechowski/react-social-share-buttons

**Review by roughike — APPROVED** [review]

**Comment by roughike** [comment]
The build is working again and the buttons look good! 💯

**Review by ragmha — CHANGES_REQUESTED** [review]
screen shot 2018-03-15 at 12 46 28

- [ ] Make Button Size similar to Quora site:
- [ ]  Regarding spacing, you can use flexbox  for justify-content: space-between
- [ ]  Add media queries so that buttons are sized properly across different screen size 

Remaining looks great! 💯

**Review by ragmha — APPROVED** [review]

**Pull request merged**